iina/PrefCodecViewController.swift

```diff
@@ -110,6 +110,17 @@ class PrefCodecViewController: PreferenceViewController, PreferenceWindowEmbedda
     hwdecDescriptionTextField.stringValue = hwdec.localizedDescription
   }
 
+  // TODO: add "auto" option instead of 0 value
+  // TODO: add option to support "inf"
+  @IBAction func toneMappingTargetPeakAction(_ sender: NSTextField) {
+    let newValue = sender.integerValue
+    // constrain to valid mpv values
+    let validValue = newValue == 0 ? 0 : newValue.clamped(to: 10...10000)
```

Comment:
Merely constraining the value is not the convention used when the user enters invalid values for other settings. Trying entering `-1`. Other settings pop-up an alert box.

Reply:
I kinda loathe the built-in modal popup. It's a really old paradigm, it's obtrusive, and not even very helpful. It doesn't tell the user what the valid values actually are, and if they choose the `OK` button the field is allowed to be left blank, which leaves them guessing about the current value. At least the `clamped` approach chooses the closest valid value, and doesn't even require reading.

Also...I actually don't know how to make the built-in popup work for `0` because it's outside the contiguous range of 10-10,000. As far as I can tell, the popup is provided via the text field cell's `NumberFormatter`, which checks against accepted min and max only. I don't see any places to hook into or override it to provide more complex validation.

Reply:
Sorry, I didn't mean to imply we needed to use the alert provided by AppKit. I was expecting we would use `Utility.showAlert` to show our own alert with the appropriate information about what values are allowed.

I agree, the one provided by `NumberFormatter` can't be used due to the lack of a contiguous range. And yes, I too dislike the lack of info in the alert `NumberFormatter` provides.

Reply:
@svobs Still waiting for an alert to be added.

Reply:
I pushed an update which uses an alert instead. I tried to mimic Apple's alert mostly, but it has only an OK button and resets the field to the current (valid) prefs value.

Reply:
This looks good:
![alert](https://github.com/user-attachments/assets/12759c11-a381-400a-a84f-fb4ea82dccfb)